pymeasure/instruments/common_base.py

```diff
@@ -0,0 +1,474 @@
+    """
+
+    # Variable holding the list of DynamicProperty parameters that are configurable
+    # by users
+    _fget_params_list = ('get_command',
+                         'values',
+                         'map_values',
+                         'get_process',
+                         'command_process',
+                         'check_get_errors')
+
+    _fset_params_list = ('set_command',
+                         'validator',
+                         'values',
+                         'map_values',
+                         'set_process',
+                         'command_process',
+                         'check_set_errors')
+
+    # Prefix used to store reserved variables
+    __reserved_prefix = "___"
+
+    def __init__(self):
+        self._special_names = self._setup_special_names()
+
+    def _setup_special_names(self):
+        """ Return list of class/instance special names.
+
+        Compute the list of special names based on the list of
+        class attributes that are a DynamicProperty. Check also for class variables
+        with special name and copy them at instance level
+        Internal method, not intended to be accessed at user level."""
+        special_names = []
+        dynamic_params = tuple(set(self._fget_params_list + self._fset_params_list))
+        # Check whether class variables of DynamicProperty type are present
+        for obj in (self,) + self.__class__.__mro__:
+            for attr_name, attr in obj.__dict__.items():
+                if isinstance(attr, DynamicProperty):
+                    special_names += [attr_name + "_" + key for key in dynamic_params]
+        # Check if special variables are defined at class level
+        for obj in (self,) + self.__class__.__mro__:
+            for attr in obj.__dict__:
+                if attr in special_names:
+                    # Copy class special variable at instance level, prefixing reserved_prefix
+                    setattr(self, self.__reserved_prefix + attr, obj.__dict__[attr])
+        return special_names
+
+    def __setattr__(self, name, value):
+        """ Add reserved_prefix in front of special variables."""
+        if hasattr(self, '_special_names'):
+            if name in self._special_names:
+                name = self.__reserved_prefix + name
+        super().__setattr__(name, value)
+
+    def __getattribute__(self, name):
+        """ Prevent read access to variables with special names used to
+        support dynamic property behaviour."""
+        if name in ('_special_names', '__dict__'):
+            return super().__getattribute__(name)
+        if hasattr(self, '_special_names'):
+            if name in self._special_names:
+                raise AttributeError(
+                    f"{name} is a reserved variable name and it cannot be read")
+        return super().__getattribute__(name)
+
+    # Channel management
+    def add_child(self, cls, id, collection="channels", prefix="ch", **kwargs):
```

Comment:
Is there a way to remove existing children? The instrument I'm working with now can change the number of channels/traces on the go (since they're all virtual), so for such an instrument it might be useful if there is a way to add and remove channels at random moments (I can imagine calling a method to change / query the number of channels on the instrument and adjust the number of `Channels` accordingly, or e.g. destroy all channels and make all new channels).

Reply:
I guess nothing stands in the way of that, but we did not have that need on our radar so far.

Reply:
Should be quite simple.
I thought about `del_child(child)` (child is the instance. This allows deleting the list entry and the attribute.

Reply:
Sounds good. `remove_child`, maybe? :-)

Reply:
It's not difficult to remove a child, however looping over a list while deleting members of that list is more complicated. Nevertheless, I'll add that `remove_child` method.